stix_shifter_modules/crowdstrike_logscale/stix_transmission/results_connector.py

```diff
@@ -0,0 +1,214 @@
+from stix_shifter_utils.modules.base.stix_transmission.base_json_results_connector import BaseJsonResultsConnector
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from stix_shifter_utils.utils import logger
+from .status_connector import StatusConnector
+from .delete_connector import DeleteConnector
+import regex
+import json
+from flatten_json import unflatten
+
+class InvalidSearchIdException(Exception):
+    pass
+
+# LogScale event fields starts with @timestamp and @error_msg will be overwritten while un-flattening.
+# These fields are excluded while un flattening
+DS_FLATTEN_KEY_EXCLUDE = ["@timestamp", "@error_msg"]
+
+
+class ResultsConnector(BaseJsonResultsConnector):
+    def __init__(self, api_client):
+        self.api_client = api_client
+        self.logger = logger.set_logger(__name__)
+        self.connector = __name__.split('.')[1]
+
+    async def create_results_connection(self, search_id, offset, length, metadata=None):
```

Comment:
I've written a very high level outline of how I'd expect this method to work.

        #I've broken this into methods to improve readability                
        #The structure of create_results_connections should look something like this
        async def create_results_connection(self, search_id, offset, length, metadata=None):
        data = self.get_data_from_response(search_id)

        next_query = self.get_next_query(metadata[query], metadata[event_id], metadata[event_timestamp], page_size, len(data))

        while(data < length):
            current_search_id = await self.api_client.create_search(next_query)
            await self.wait_for_search_completion(current_search_id)
            results = await self.get_results_from_search(current_search_id)
            self.add_results_to_data(data, results)
            next_query = self.get_next_query(metadata[query], metadata[event_id], metadata[event_timestamp], page_size, len(data))
        
        self.prepare_meta_data(metadata)
        return return_obj`

The actual implementation may require more nuance (the actual implementation needs to account for more), but the general logic should look like this.

Reply:
updated as per suggestions